### PR TITLE
Add linchpin mock integration tests to aws provider 

### LINKED ISSUE #1545: Replace Deprecated module used for aws_ec2 network
**Describe the bug**
The following message is seen while running all the aws_ec2 network related modules. 
[DEPRECATION WARNING]: ec2_vpc_net_facts is kept for backward compatibility but usage is discouraged. The module documentation details page may explain more about this rationale.. This feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.

The deprecated modules must be replaced with the newer module.

### PR DESCRIPTION
**Pull request by samvarankashyap**
$subject
fixes #1545 #1538

### CODE REVIEW

**Comment by samvarankashyap** [comment]
@14rcole Needs review

File: linchpin/provision/action_plugins/ec2_vpc_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        Simple action plugin that returns the mocked output
+        when linchpin_mock is True
+        """
+        super(ActionModule, self).run(tmp, task_vars)
+        # contains all the module arguments
+        module_args = self._task.args.copy()
+        # task vars.keys() contains all the variable  required
+        # when passed a extra_var as key value pair task_vars
+        # would return mocked output of the named module.
+        # print(task_vars['vars'].keys())
```
Comment (14rcole) [inline linchpin/provision/action_plugins/ec2_vpc_endpoint.py]: Can you delete these?  They're in a couple of the other files too

Comment (samvarankashyap) [reply linchpin/provision/action_plugins/ec2_vpc_endpoint.py]: @14rcole Sure I will update the PR

File: linchpin/FilterUtils/FilterUtils.py [hunk 2]
```diff
@@ -287,10 +287,12 @@ def get_os_server_names(topo_output):
     return names
 
 
-def write_to_file(data, path, filename):
+def write_to_file(data, path, filename, extn='.output'):
+    
     filename = filename.replace(' ', '_').lower()
     path = os.path.expanduser(path)
-    fd = open(path + filename + ".output", "w")
+    print(path)
```
Comment (14rcole) [inline linchpin/FilterUtils/FilterUtils.py]: Unnecessary print statement?

File: linchpin/provision/action_plugins/ec2.py [hunk 3]
```diff
@@ -69,6 +84,7 @@ def set_dict(context, key, value):
         done = False
         old_state = ''
         while not done:
+            print(result)
```
Comment (14rcole) [inline linchpin/provision/action_plugins/ec2.py]: print statement?

**Comment by samvarankashyap** [comment]
@14rcole I will make the changes

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**